tests/cpp/test_combine_mul_sum.cpp

```diff
@@ -269,6 +271,341 @@ TEST_F(CombineMulSumAsMmaTest, UseMatmulScheduler) {
+        executor_cache.getMostRecentKernelRuntime();
+    ASSERT_NE(runtime, nullptr);
+
+    if (expect_aten_eval) {
+      EXPECT_TRUE(runtime->isSegmented());
+    } else {
+      EXPECT_FALSE(runtime->isSegmented());
+    }
+
+    ScheduleHeuristic heuristic =
+        runtime->schedulerHeuristics()->heuristicsList().front()->heuristic();
+    if (expect_aten_eval) {
+      EXPECT_EQ(heuristic, ScheduleHeuristic::ExprEval);
+    } else {
+      // Ensure that the Matmul scheduler ran.
+      // Assert here since we will inspect the kernel next, which we can't
+      // do if ExprEval accepts the segment.
+      ASSERT_EQ(heuristic, ScheduleHeuristic::Matmul);
+      // Ensure there's an MmaOp.
+      EXPECT_FALSE(
+          ir_utils::getOpsOfType<MmaOp>(runtime->executors().at(0).kernel())
+              .empty());
+    }
+
+    testValidate(
+        executor_cache.fusion(), outputs, inputs, {tref}, __LINE__, __FILE__);
+  };
+  // CombineMulSumAsMmaTest disabled MatmulExprEval, but we need it
+  // enabled
+  DisableOptionsGuard dog;
+  DisableOptionsGuard::getCurOptions().unset(DisableOption::MatmulExprEval);
+  EnableOptionsGuard eog;
+
+  // Run the test with and without matmul fusion enabled
+  EnableOptionsGuard::getCurOptions().unset(EnableOption::FuseMatmul);
+  run(2, 2, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/true);
+  run(2, 2, -1, /*transpose_a_alloc=*/true, /*expect_aten_eval=*/true);
+
+  // Allow Matmul Scheduler to fuse MatmulOp and LinearOp
+  EnableOptionsGuard::getCurOptions().set(EnableOption::FuseMatmul);
+  run(2, 2, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/false);
+  // We cannot yet handle allocation domain in matmul scheduler
+  run(2, 2, -1, /*transpose_a_alloc=*/true, /*expect_aten_eval=*/true);
+
+  // Don't fuse 1D inputs
+  run(1, 2, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/true);
+  run(2, 1, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/true);
+  // TODO: The following currently fails but it should not be translated to
+  // LinearOp to begin with
+  // run(1, 1, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/true);
+
+  // Batch dims in input
+  // TODO: mixed length inputs via broadcasted batch dims
+  // We currently reject differently-sized inputs since these translate to
+  // multiple M or N dims
+  run(3, 2, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/true);
+  // TODO: We don't yet support multiple batch dims in matmul scheduler
+  run(4, 2, -1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/true);
+
+  // Bias cases
+  run(2, 2, 0, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/false);
+  run(2, 2, 1, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/false);
+  // Undocumented 2D bias support
+  // TODO: Currently failing in propagateBoundValuesThroughExactMaps, indicating
+  // possible PairwiseRootDomainMap issue for 2D bias?
+  // run(2, 2, 2, /*transpose_a_alloc=*/false, /*expect_aten_eval=*/false);
```

Comment:
I'm currently investigating this issue.

Reply:
We do not support 2D bias. Thunder does not allow 2D bias either.

https://github.com/NVIDIA/Fuser/pull/2240#discussion_r1603803176

Reply:
We will need to update the mapping if we want to support 2D bias, but that case is tricky, since for this case as well, we have 2D bias but the mapping is not to [M, N]: `A = [2, 4, 6], B [3, 6], Bias = [2*4, 3] `

Reply:
OK I remember that now. In that case, with your permission I'll just add a check to our `linear` op so that we fail earlier, and I'll remove this test case.